tests/phpunit/tests/media.php

```diff
@@ -3710,6 +3710,98 @@ function() {
 		$this->assertSame( 3, $omit_threshold );
 	}
 
+	/**
+	 * Tests that wp_get_loading_attr_default() returns the expected loading attribute value before loop.
+	 *
+	 * @ticket 58211
+	 *
+	 * @covers ::wp_get_loading_attr_default
+	 *
+	 * @dataProvider data_wp_get_loading_attr_default_before_and_no_loop
+	 *
+	 * @param string $context
+	 */
+	public function test_wp_get_loading_attr_default_before_loop( $context ) {
+		global $wp_query, $wp_the_query;
+
+		$wp_query = new WP_Query( array( 'post__in' => array( self::$post_ids['publish'] ) ) );
+		$this->reset_content_media_count();
+		$this->reset_omit_loading_attr_filter();
+
+		// Lazy if not main query.
+		$this->assertSame( 'lazy', wp_get_loading_attr_default( $context ) );
+
+		// Set current query as main query.
+		$wp_the_query = $wp_query;
+
+		// For main query return false for any instance before end of get_header.
```

Comment:
See above, that was probably a bit unclear on the ticket: It should return false after the `get_header` action has fired - when it's running still, we can also return `false`.
```suggestion
		// For main query return false for any image after the 'get_header' action.
```